File: qooxdoo-contrib/QWT/trunk/toolkit/compiler/src/test/resources/type/extends.java
package type;

class Base {
    int b;
    
    public int getB() {
        return b;
    }
    public int fn() {
        return b * b;
    }
}

class Ext extends Base {
    long e;
    
    public Ext() {
        this(0);
    }
    public Ext(long e) {
        super();
        this.e = e;
    }
    public int get() {
        return getB();
    }
    
    public int fn() {
        return super.fn() + 5;
    }
    public static Ext create() {
        return new Ext();
    }
}

File: trunk/qooxdoo-contrib/QWT/tags/0.7.2/toolkit/compiler/src/test/resources/type/extends.js
touchPackage("type");
type.Base=function() {
}
defineClass(type.Base, 'type.Base', java.lang.Object);
type.Base.prototype.b = 0;
type.Base.prototype.getB = function() {
    return this.b;
}
type.Base.prototype.fn = function() {
    return this.b * this.b;
}
touchPackage("type");
type.Ext=function() {
}
defineClass(type.Ext, 'type.Ext', type.Base);
type.Ext.prototype.e = 0;
type.Ext.init = function() {
    type.Ext.init1.call(this, 0);
}
type.Ext.init1 = function(e) {
    type.Base.call(this);
    this.e = e;
}
type.Ext.prototype.get = function() {
    return this.getB();
}
type.Ext.prototype.fn = function() {
    return this.constructor.superclass.prototype.fn.call(this) + 5;
}
type.Ext.create = function() {
    return newObject(type.Ext, type.Ext.init, []);
}
